xbmc/addons/AddonManager.cpp

```diff
@@ -734,10 +734,34 @@ bool CAddonMgr::ReloadAddon(AddonPtr& addon)
   if (!addon ||!m_cpluff || !m_cp_context)
     return false;
 
-  m_cpluff->uninstall_plugin(m_cp_context, addon->ID().c_str());
-  return FindAddons()
-      && GetAddon(addon->ID(), addon, ADDON_UNKNOWN, false)
-      && EnableAddon(addon->ID());
+  AddonPtr tmp;
```

Comment:
Because `addon` is the out parameter of this method and it will be the new instance after reload, so if you use it here and it failed (and the caller doesn’t check the return status) it will have the wrong one (maybe it doesn't really matter as you always *should* check the return value)